### ISSUE TITLE
SQL errors due to unquoted table names

### ISSUE THREAD

**Issue by robrecord** [kylephillips/wp-nested-pages#208]
I get the following error upon using the plugin (listing nested pages) with my custom post type:

> **WordPress database error**: [You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '`-level,GROUP_CONCAT(DISTINCT t_course-degree-type.term_id SEPARATOR ',') AS cour`' at line 1]

I believe this is because the table names are not being quoted with backticks.

The unexpanded code is here:

```SQL
SELECT DISTINCT wp_uv9fkd_posts.*,GROUP_CONCAT(DISTINCT t_course-level.term_id SEPARATOR ',') AS course-level,GROUP_CONCAT(DISTINCT t_course-degree-type.term_id SEPARATOR ',') AS course-degree-type,GROUP_CONCAT(DISTINCT t_course-subject.term_id SEPARATOR ',') AS course-subject,GROUP_CONCAT(DISTINCT t_genre.term_id SEPARATOR ',') AS genre FROM wp_uv9fkd_posts LEFT JOIN wp_uv9fkd_term_relationships AS tr_course-level ON tr_course-level.object_id = wp_uv9fkd_posts.ID LEFT JOIN wp_uv9fkd_term_taxonomy tt_course-level ON tt_course-level.term_taxonomy_id = tr_course-level.term_taxonomy_id AND tt_course-level.taxonomy = 'course-level' LEFT JOIN wp_uv9fkd_terms AS t_course-level ON t_course-level.term_id = tt_course-level.term_id LEFT JOIN wp_uv9fkd_term_relationships AS tr_course-degree-type ON tr_course-degree-type.object_id = wp_uv9fkd_posts.ID LEFT JOIN wp_uv9fkd_term_taxonomy tt_course-degree-type ON tt_course-degree-type.term_taxonomy_id = tr_course-degree-type.term_taxonomy_id AND tt_course-degree-type.taxonomy = 'course-degree-type' LEFT JOIN wp_uv9fkd_terms AS t_course-degree-type ON t_course-degree-type.term_id = tt_course-degree-type.term_id LEFT JOIN wp_uv9fkd_term_relationships AS tr_course-subject ON tr_course-subject.object_id = wp_uv9fkd_posts.ID LEFT JOIN wp_uv9fkd_term_taxonomy tt_course-subject ON tt_course-subject.term_taxonomy_id = tr_course-subject.term_taxonomy_id AND tt_course-subject.taxonomy = 'course-subject' LEFT JOIN wp_uv9fkd_terms AS t_course-subject ON t_course-subject.term_id = tt_course-subject.term_id LEFT JOIN wp_uv9fkd_term_relationships AS tr_genre ON tr_genre.object_id = wp_uv9fkd_posts.ID LEFT JOIN wp_uv9fkd_term_taxonomy tt_genre ON tt_genre.term_taxonomy_id = tr_genre.term_taxonomy_id AND tt_genre.taxonomy = 'genre' LEFT JOIN wp_uv9fkd_terms AS t_genre ON t_genre.term_id = tt_genre.term_id WHERE 1=1 AND wp_uv9fkd_posts.post_type = 'course' AND ((wp_uv9fkd_posts.post_status = 'publish' OR wp_uv9fkd_posts.post_status = 'future' OR wp_uv9fkd_posts.post_status = 'draft' OR wp_uv9fkd_posts.post_status = 'pending' OR wp_uv9fkd_posts.post_status = 'trash' OR wp_uv9fkd_posts.post_status = 'private')) GROUP BY wp_uv9fkd_posts.ID ORDER BY wp_uv9fkd_posts.menu_order, wp_uv9fkd_posts.post_date DESC
```

which expands to this (note the spacing in the middle of table names, I believed this is caused by the dash in the table name):

```SQL
SELECT DISTINCT wp_uv9fkd_posts.*,
                Group_concat(DISTINCT t_course-level.term_id SEPARATOR ',')       AS course-level,
                group_concat(DISTINCT t_course-degree-type.term_id SEPARATOR ',') AS course-degree-type,
                group_concat(DISTINCT t_course-subject.term_id SEPARATOR ',')     AS course-subject,
                group_concat(DISTINCT t_genre.term_id SEPARATOR ',')              AS genre
FROM            wp_uv9fkd_posts
LEFT JOIN       wp_uv9fkd_term_relationships AS tr_course-level
ON              tr_course                                -level.object_id = wp_uv9fkd_posts.id
LEFT JOIN       wp_uv9fkd_term_taxonomy tt_course        -level
ON              tt_course                                -level.term_taxonomy_id = tr_course-level.term_taxonomy_id
AND             tt_course                                -level.taxonomy = 'course-level'
LEFT JOIN       wp_uv9fkd_terms AS t_course              -level
ON              t_course                                 -level.term_id = tt_course-level.term_id
LEFT JOIN       wp_uv9fkd_term_relationships AS tr_course-degree-type
ON              tr_course                                -degree-type.object_id = wp_uv9fkd_posts.id
LEFT JOIN       wp_uv9fkd_term_taxonomy tt_course        -degree-type
ON              tt_course                                -degree-type.term_taxonomy_id = tr_course-degree-type.term_taxonomy_id
AND             tt_course                                -degree-type.taxonomy = 'course-degree-type'
LEFT JOIN       wp_uv9fkd_terms AS t_course              -degree-type
ON              t_course                                 -degree-type.term_id = tt_course-degree-type.term_id
LEFT JOIN       wp_uv9fkd_term_relationships AS tr_course-subject
ON              tr_course                                -subject.object_id = wp_uv9fkd_posts.id
LEFT JOIN       wp_uv9fkd_term_taxonomy tt_course        -subject
ON              tt_course                                -subject.term_taxonomy_id = tr_course-subject.term_taxonomy_id
AND             tt_course                                -subject.taxonomy = 'course-subject'
LEFT JOIN       wp_uv9fkd_terms AS t_course              -subject
ON              t_course                                 -subject.term_id = tt_course-subject.term_id
LEFT JOIN       wp_uv9fkd_term_relationships AS tr_genre
ON              tr_genre.object_id = wp_uv9fkd_posts.id
LEFT JOIN       wp_uv9fkd_term_taxonomy tt_genre
ON              tt_genre.term_taxonomy_id = tr_genre.term_taxonomy_id
AND             tt_genre.taxonomy = 'genre'
LEFT JOIN       wp_uv9fkd_terms AS t_genre
ON              t_genre.term_id = tt_genre.term_id
WHERE           1=1
AND             wp_uv9fkd_posts.post_type = 'course'
AND             ((
                                                wp_uv9fkd_posts.post_status = 'publish'
                                OR              wp_uv9fkd_posts.post_status = 'future'
                                OR              wp_uv9fkd_posts.post_status = 'draft'
                                OR              wp_uv9fkd_posts.post_status = 'pending'
                                OR              wp_uv9fkd_posts.post_status = 'trash'
                                OR              wp_uv9fkd_posts.post_status = 'private'))
GROUP BY        wp_uv9fkd_posts.id
ORDER BY        wp_uv9fkd_posts.menu_order,
                wp_uv9fkd_posts.post_date DESC
```

**Comment by kylephillips** [comment]
I'm having trouble reproducing this error, but multiple people are reporting it.

Could you try the attached patched version of the plugin?

If this does not resolve the issue, would you let me know what mysql version you are using?
[wp-nested-pages.zip](https://github.com/kylephillips/wp-nested-pages/files/1277825/wp-nested-pages.zip)

**Comment by kylephillips** [comment]
Updated possible fix to try:
https://github.com/kylephillips/wp-nested-pages/releases/tag/2.0.1.1

**Comment by kylephillips** [comment]
This should be resolved in v2.0.2
